Shopify/checkout-sheet-kit-android
# Update FileChooserResultContract.kt

This pull request resolves issues with handling camera image results across all devices, particularly targeting inconsistencies found on certain devices like Samsung and Xiaomi during checkout.

Note: I have made changes in the sample app not in the lib so thats why, I haven't updated the ReadMe and ChangeLog files

### What changes are you making?

- Improved handling of camera image results for all devices by ensuring the correct URI is received, addressing device-specific issues.
- Adjusted camera and file picker integration for consistent behaviour across different Android manufacturers.

### How to test

- Launch the checkout flow that includes the camera and file picker feature.
- Test capturing an image using the camera on a variety of devices (e.g., Samsung, Xiaomi).
- Verify that the captured image is returned correctly across all tested devices.
- Test the fallback to the file picker to ensure proper handling of images from the gallery.

---

### Before you merge

> [!IMPORTANT]
>
> - [ ] I've added tests to support my implementation
> - [x] I have read and agree with the [Contribution Guidelines](https://github.com/shopify/checkout-sheet-kit-android/blob/main/.github/CONTRIBUTING.md).
> - [x] I have read and agree with the [Code of Conduct](https://github.com/shopify/checkout-sheet-kit-android/blob/main/.github/CODE_OF_CONDUCT.md).
> - [ ] I've updated the [README](https://github.com/shopify/checkout-sheet-kit-android).

---

<details>
<summary>Checklist for releasing a new version</summary>


</details>

> [!TIP]
> See the [Contributing documentation](https://github.com/shopify/checkout-sheet-kit-android/blob/main/.github/CONTRIBUTING.md#releasing-a-new-version) for instructions on how to publish a new version of the library.
